src/format.rs

```diff
@@ -640,6 +651,71 @@ impl<'a> Formatter<'a> {
         }
     }
 
+    fn segment(&mut self, segment: &UntypedExprBinSegment) -> Document {
+        match segment {
+            UntypedExprBinSegment { value, options, .. } if *options == [] => self.expr(&value),
+
+            UntypedExprBinSegment { value, options, .. } => {
+                // TODO: Is there a better option than these self.clone() calls?
+                let options = options.iter().map(|o| self.clone().segment_option(o));
```

Comment:
Why is it that the formatter is cloned here and on line 662?

Reply:
Good question. I don't remember to be honest, it must have refused to compile for reasons I couldn't be bothered to properly figure out at the time, hence the quick fix and the TODOs. It will be sorted before this becomes a real PR :)

Reply:
Thanks!